File: app/src/main/java/com/slidely/photoframe/features/preferences/PreferencesFragment.java
package com.slidely.photoframe.features.preferences;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.EditTextPreference;
import androidx.preference.PreferenceFragmentCompat;

import com.slidely.photoframe.R;

public class PreferencesFragment extends PreferenceFragmentCompat {
    public static final String SPEED_KEY = "speed";

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        //noinspection ConstantConditions
        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        addPreferencesFromResource(R.xml.preferences);
        EditTextPreference speed = (EditTextPreference) findPreference(SPEED_KEY);
        speed.setSummary(speed.getText());
        speed.setOnPreferenceChangeListener((preference, newValue) -> {
            preference.setSummary((CharSequence) newValue);
            return true;
        });
    }
}

File: app/src/main/java/com/k4dima/photoframe/features/preferences/PreferencesFragment.kt
package com.k4dima.photoframe.features.preferences

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.EditTextPreference
import androidx.preference.PreferenceFragmentCompat
import com.k4dima.photoframe.R
import com.k4dima.photoframe.features.main.data.DefaultPreferenceRepository

class PreferencesFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        (activity as AppCompatActivity).supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        addPreferencesFromResource(R.xml.preferences)
        (findPreference(DefaultPreferenceRepository.speedKey(context!!)) as EditTextPreference).apply {

            val seconds = " seconds"
            summary = text + seconds
            setOnPreferenceChangeListener { preference, newValue ->
                preference.summary = newValue as String + seconds
                true
            }
        }
    }
}